coderd/workspacebuilds.go

```diff
@@ -948,33 +948,17 @@ func (api *API) convertWorkspaceBuild(
 }
 
 func convertWorkspaceResource(resource database.WorkspaceResource, agents []codersdk.WorkspaceAgent, metadata []database.WorkspaceResourceMetadatum) codersdk.WorkspaceResource {
-	metadataMap := map[string]database.WorkspaceResourceMetadatum{}
-
-	// implicit metadata fields come first
-	metadataMap["type"] = database.WorkspaceResourceMetadatum{
-		Key:       "type",
-		Value:     sql.NullString{String: resource.Type, Valid: true},
-		Sensitive: false,
-	}
-	// explicit metadata fields come afterward, and can override implicit ones
-	for _, field := range metadata {
-		metadataMap[field.Key] = field
-	}
-
+	slices.SortFunc(metadata, func(a, b database.WorkspaceResourceMetadatum) bool {
+		return a.ID < b.ID
+	})
```

Comment:
Why sort in Go when we get them back from Postgres already sorted?

Reply:
I wasn't sure if we would, but if we do then I'll rm this.